### PR TITLE
Transport fixes and cleanup

### PR DESCRIPTION
**Pull request by ds58**

### CODE REVIEW

File: ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java [hunk 1]
```diff
@@ -136,112 +136,136 @@ public ParticipantProfile discoveryServer(String discoveryServerAddress, int dis
...
-    * 
-    * Functionality, this will create a new UDPv4 transport with the whitelist set to "bindToAddressRestrictions". Optionally, a shared memory transport will be added as well.
-    * useBuiltinTransports will be set to false 
-    * 
-    * @param addSharedMemoryTransport Enabled shared memory communication by adding a shared memory transport to this participant.
-    * @param bindToAddressRestrictions Limit the scope of this participant to the list of hosts. If null or empty, the participant will not be able to communicate.
-    * @return
+    * Add transport to use with this participant and register it to this participant
     */
-   public ParticipantProfile bindToAddressRestrictions(boolean addSharedMemoryTransport, List<InetAddress> bindToAddressRestrictions)
```
Comment (ds58) [inline ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java:148]: I removed `bindToAddressRestrictions`. It was not used anywhere and it is essentially duplicate functionality to what is now in `addUDPv4Transport`

File: ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java [hunk 2]
```diff
@@ -35,10 +35,10 @@ public ParticipantProfile()
       // Create default elements for participant profile
       BuiltinAttributesType builtin = new BuiltinAttributesType();
       DiscoverySettingsType discoverySettingsType = new DiscoverySettingsType();
+      builtin.setDiscoveryConfig(discoverySettingsType);
```
Comment (ds58) [inline ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java:38]: The ordering here was sketchy

File: ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java [hunk 3]
```diff
@@ -136,112 +136,136 @@ public ParticipantProfile discoveryServer(String discoveryServerAddress, int dis
...
-    * Functionality, this will create a new UDPv4 transport with the whitelist set to "bindToAddressRestrictions". Optionally, a shared memory transport will be added as well.
-    * useBuiltinTransports will be set to false 
-    * 
-    * @param addSharedMemoryTransport Enabled shared memory communication by adding a shared memory transport to this participant.
-    * @param bindToAddressRestrictions Limit the scope of this participant to the list of hosts. If null or empty, the participant will not be able to communicate.
-    * @return
+    * Add transport to use with this participant and register it to this participant
     */
-   public ParticipantProfile bindToAddressRestrictions(boolean addSharedMemoryTransport, List<InetAddress> bindToAddressRestrictions)
+   public ParticipantProfile addTransport(TransportDescriptorType transport)
```
Comment (ds58) [inline ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java:141]: See this example for understanding the difference between transport_descriptors and userTransports
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<dds xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <profiles>
        <transport_descriptors>
            <transport_descriptor>
                <transport_id>dc261415-468c-4302-9da6-f1e3ead3c9a3</transport_id>
                <type>SHM</type>
            </transport_descriptor>
            <transport_descriptor>
                <transport_id>41087b0d-7915-4c1d-a276-57de9a749fea</transport_id>
                <type>UDPv4</type>
            </transport_descriptor>
        </transport_descriptors>
        <participant profile_name="66a19794-8016-4736-b280-32930be03091">
            <domainId>219</domainId>
            <rtps>
                <name>srt_stream_image_message_republisher</name>
                <builtin>
                    <discovery_config>
                        <leaseDuration>
                            <nanosec>-1</nanosec>
                            <sec>2147483647</sec>
                        </leaseDuration>
                    </discovery_config>
                </builtin>
                <userTransports>
                    <transport_id>dc261415-468c-4302-9da6-f1e3ead3c9a3</transport_id>
                    <transport_id>41087b0d-7915-4c1d-a276-57de9a749fea</transport_id>
                </userTransports>
                <useBuiltinTransports>false</useBuiltinTransports>
            </rtps>
        </participant>
    </profiles>
</dds>
```

File: ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java [hunk 4]
```diff
@@ -136,112 +136,136 @@ public ParticipantProfile discoveryServer(String discoveryServerAddress, int dis
...
+
       addTransport(transportDescriptor);
-      
+
       return this;
    }
-   
-   public ParticipantProfile useBuiltinTransports(boolean useBuiltinTransports)
+
+   public ParticipantProfile addUDPv4Transport(InetAddress... addressRestriction)
```
Comment (ds58) [inline ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java:195]: I added this method, the logic was duplicated in a couple places. It adds a UDPv4 transport with optional restrictions on addresses

File: ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java [hunk 5]
```diff
@@ -136,112 +136,136 @@ public ParticipantProfile discoveryServer(String discoveryServerAddress, int dis
...
+
+         udp4Transport.setInterfaceWhiteList(addressWhitelist);
+      }
+
+      addTransport(udp4Transport);
+
       return this;
    }
-   
-   public boolean isUseBuiltinTransports()
```
Comment (ds58) [inline ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java:224]: Moved to below where it makes more sense

File: ros2-library/src/main/java/us/ihmc/ros2/ROS2Node.java [hunk 6]
```diff
@@ -609,53 +603,27 @@ public void destroy()
...
-
-         for (InetAddress addr : addressRestriction)
-         {
-            JAXBElement<String> addressElement = new JAXBElement<>(new QName(FastRTPSDomain.FAST_DDS_XML_NAMESPACE, "address"), String.class, addr.getHostAddress());
-            addressWhitelist.getAddressOrInterface().add(addressElement);
-         }
-
-         transportDescriptor.setInterfaceWhiteList(addressWhitelist);
-      }
-
```
Comment (ds58) [inline ros2-library/src/main/java/us/ihmc/ros2/ROS2Node.java:645]: Removed logic that is now in `ParticipantProfile#addUDPv4Transport`

**Review by calvertdw — APPROVED** [review]

**Pull request merged**